ironfish/src/rpc/routes/chain/getTransaction.ts

```diff
@@ -72,36 +86,44 @@ router.register<typeof GetTransactionRequestSchema, GetTransactionResponse>(
     }
     const hashBuffer = BlockHashSerdeInstance.deserialize(request.data.blockHash)
 
-    const block = await node.chain.getBlock(hashBuffer)
-    if (!block) {
+    const blockHeader = await node.chain.getHeader(hashBuffer)
+    if (!blockHeader) {
       throw new ValidationError(`No block found`)
     }
 
     // Empty response used for case that transaction not found
     const rawTransaction: GetTransactionResponse = {
       fee: '0',
       expiration: 0,
+      index: 0,
       notesCount: 0,
       spendsCount: 0,
       signature: '',
       notesEncrypted: [],
       mints: [],
       burns: [],
     }
+    const transactions = await node.chain.getBlockTransactions(blockHeader)
 
-    block.transactions.map((transaction) => {
+    transactions.map(({ transaction, initialNoteIndex }) => {
       if (transaction.hash().toString('hex') === request.data.transactionHash) {
         const fee = transaction.fee().toString()
         const expiration = transaction.expiration()
         const signature = transaction.transactionSignature()
         const notesEncrypted = []
 
+        let noteIndex = initialNoteIndex
         for (const note of transaction.notes) {
-          notesEncrypted.push(note.serialize().toString('hex'))
+          notesEncrypted.push({
+            noteIndex: noteIndex,
```

Comment:
We can just store this on the transaction and not duplicate it, since it can be calculated by the consumer of this.

Reply:
I have tried this way. Store the transaction initialNoteIndex and then calculation the noteIndex with the initialNoteIndex and transaction notes. You know, the explorer backend database can do this, but i did get some bad errors. But idk whether that would happen with node.

Reply:
What type of errors were you getting? I would be concerned if transactions come out of order or something

Reply:
As you say, transactions come out of order, then this transaction order and block header `treeSize` used in calculation of transaction `initialNoteIndex`, `note index`. Then wrong `noteWitness` to this `note` is used in transaction creation and this transaction creation failed (transaction proof verification failed).